src/diffusers/models/modeling_utils.py

```diff
@@ -305,7 +330,14 @@ def save_pretrained(
 
         os.makedirs(save_directory, exist_ok=True)
 
-        model_to_save = self
+        if push_to_hub:
+            commit_message = kwargs.pop("commit_message", None)
+            repo_id = kwargs.pop("repo_id", save_directory.split(os.path.sep)[-1])
+            repo_id = self._create_repo(repo_id, **kwargs)
+            files_timestamps = self._get_files_timestamps(save_directory)
+
+        # Only save the model itself if we are using distributed training
+        model_to_save = unwrap_model(self)
```

Comment:
```suggestion
```

I'm in favor of not adding this for now because:
a) We can unwrap in the model scripts right away & for now I think it's easier to not promise the user that we'll unwrap the model automatically. Instead for now I'd force the user to just unwrap the model themselves
b) It would not work with torch.compiled models